packages/typescript-resolver-files-e2e/src/test-extended-object-types/schema-federated-extended-object-type/schema/topic/topic.graphqls

```diff
@@ -0,0 +1,63 @@
+extend type Query {
+  topicById(id: ID!): TopicByIdPayload!
+  topicsCreatedByUser(
+    input: TopicsCreatedByUserInput!
+  ): TopicsCreatedByUserPayload!
+}
+
+extend type Mutation {
+  topicCreate(input: TopicCreateInput!): TopicCreatePayload!
+  topicEdit(input: TopicEditInput!): TopicEditPayload!
+}
+
+type Topic {
+  id: ID!
+  name: String!
+  url: String
+  creator: User!
+  createdAt: DateTime!
+}
+
+type ForeignType @key(fields: "id") {
+  id: ID!
+  topics: [Topic!]!
+}
```

Comment:
I haven't used federation in practice so I'll lean into your expertise here for these federation questions @wilhelmeek 🙏:

1. Is there a case where there's only `type ForeignType` and no `extend type ForeignType`? e.g.
2. What happens if there's a `@key` in both `type ForeignType` and `extend type ForeignType` ? What if the `fields` value is different e.g.

```graphql
type ForeignType @key(fields: "id")

extend type ForeignType @key(fields: "user") { 👈 Is this unexpected/not allowed?
    users: [User!]!
  }
```

Reply:
Hi @eddeee888, thanks for having a look at this! We are only just starting to federate ourselves so I am by no means an expert. However I believe:

> Is there a case where there's only type ForeignType and no extend type ForeignType? 

This is a valid case and should work with the current implementation. `__resolveReference` is present in the resulting type and since we are not using `Pick` it is not excluded.

> What happens if there's a @key in both type ForeignType and extend type ForeignType ? What if the fields value is different e.g.

This should also be valid via repeated directives. The reference resolver for `ForeignType` in this case would receive both id and users. It should be equivalent to:
```graphql
type ForeignType @key(fields: "id") @key(fields: "users")
```

Reply:
The typing on that second one is a bit weird, but I believe that's a separate matter.